sample-app/src/components/InputDropdown.tsx

```diff
@@ -97,24 +131,43 @@ export default function InputDropdown({
       evt.preventDefault();
     }
 
+    const isFirstSectionFocused = focusedSectionIndex === 0;
+    const isLastSectionFocused = focusedSectionIndex === options.length - 1;
     const isFirstOptionFocused = focusedOptionIndex === 0;
-    const isLastOptionFocused = focusedOptionIndex === options.length - 1;
+    const isLastOptionFocused = 
+      focusedSectionIndex !== undefined &&
+      focusedOptionIndex === options[focusedSectionIndex].results.length - 1;
     if (evt.key === 'Enter') {
-      updateInputValue(inputValue);
-      onSubmit(inputValue);
-      dispatch({ type: 'HideOptions' });
+      if (!onlySubmitOnOption || focusedOptionIndex !== undefined) {
+        onSubmit(inputValue, focusedSectionIndex, focusedOptionIndex);
+        dispatch({ type: 'HideOptions' });
+      }
     } else if (evt.key === 'Escape' || evt.key === 'Tab') {
       dispatch({ type: 'HideOptions' });
-    } else if (evt.key === 'ArrowDown' && options.length > 0 && !isLastOptionFocused) {
-      const newIndex = focusedOptionIndex !== undefined ? focusedOptionIndex + 1 : 0;
-      dispatch({ type: 'FocusOption', newIndex });
-      const newValue = options[newIndex]?.value;
+    } else if (evt.key === 'ArrowDown' && options.length > 0 && !(isLastSectionFocused && isLastOptionFocused)) {
```

Comment:
do we still need to check for `options.length > 0` if we also check for !(isLastSectionFocused && isLastOptionFocused)?

Reply:
yes, because otherwise if both focused indexes are undefined, !(isLastSectionFocused && isLastOptionFocused) would be true, and it would try to arrow down to an option that doesn't exist.

Reply:
can we default the indexes to 0?

Reply:
gotcha